CoreConversationView/src/au/gov/asd/tac/constellation/views/conversationview/ConversationBox.java

```diff
@@ -215,23 +225,72 @@ public ConversationBox(final Conversation conversation) {
             }
         });
 
-        // Create the bubbles pane
+        // Create the bubbles pane.
         bubbles = new ListView<>();
         bubbles.setStyle(JavafxStyleManager.CSS_BACKGROUND_COLOR_TRANSPARENT);
         bubbles.setCellFactory(callback -> new BubbleCell());
         VBox.setVgrow(bubbles, Priority.ALWAYS);
 
-        // Hook up the bubbles pane to the conversation
+        // Hook up the bubbles pane to the conversation.
         final ObservableList<ConversationMessage> messages = FXCollections.observableArrayList();
         bubbles.setItems(messages);
         conversation.setResultList(messages);
 
-        content.getChildren().addAll(optionsPane, contributionsPane, bubbles);
+        // Create a text field for search within bubbles.
+        searchBubbleTextField.setPromptText("Type to search for a text...");
+        searchBubbleTextField.setVisible(true);
+        searchBubbleTextField.setStyle("-fx-prompt-text-fill: #868686;");
+        searchBubbleLabel.setText("Found: " + searchHits);
+        searchBubbleLabel.setStyle("-fx-text-fill: red; -fx-effect: dropshadow(gaussian, black, 5.0, 0.0, 0.0, 0.0);");
+        searchBubbleLabel.setPadding(new Insets(4,8,4,8));
+        searchBubbleVBox.getChildren().addAll(searchBubbleTextField, searchBubbleLabel);
+
+        final ConversationSearchRefresh conversationSearchRefresh = new ConversationSearchRefresh(conversation);
+
+        // Use a dummy provider name to get the toggle pane state remain the same during the search.
+        searchBubbleTextField.addEventHandler(KeyEvent.KEY_TYPED, e -> {
+            conversationSearchRefresh.updateContributionProviderRefresh("Refresh");
+        });
+        
+        searchBubbleTextField.setOnKeyTyped(e -> {
+            searchHits = 0;
+            searchHits = SearchText.getSearchHits(conversation.getVisibleMessages());
+            searchBubbleLabel.setText("Found: " + searchHits);
+            if (searchHits > 0) {
+                searchBubbleLabel.setStyle("-fx-text-fill: yellow; -fx-effect: dropshadow(gaussian, black, 5.0, 0.0, 0.0, 0.0);");
+            } else {
+                searchBubbleLabel.setStyle("-fx-text-fill: red; -fx-effect: dropshadow(gaussian, black, 5.0, 0.0, 0.0, 0.0);");
+            }
+        });
+
+        // This code below enables the graph in sync with the search results.
+        searchBubbleTextField.textProperty().addListener((ov, oldValue, newValue) -> {
+            final Graph graph = conversation.getGraphUpdateManager().getActiveGraph();
+            ReadableGraph readableGraph = graph.getReadableGraph();
+            try {
```

Comment:
Can you convert this to a try-with-resources? This would remove the need for the finally block